security/pkg/pki/ra/k8s_ra_test.go

```diff
@@ -92,14 +92,19 @@ func TestK8sSignWithMeshConfig(t *testing.T) {
 			updatedRootCertForMeshConfig: TestCACertFile,
 			expectedFail:                 true,
 		},
+		{
+			name:      "Root cert is not specified in mesh config and cert chain contains only intermediate CA(only leaf cert + intermediate CA)",
+			certChain: path.Join(env.IstioSrc, "samples/certs", "workload-foo-cert.pem"),
+		},
+		{
+			name:         "Root cert is not specified in mesh config and Root cert from cert chain is empty(only one leaf cert)",
+			certChain:    path.Join(env.IstioSrc, "samples/certs", "cert-chain.pem"),
```

Comment:
This cert-chain is also a CA type cert which can not test the failure scenario fixed in this PR,  it is better to get a cert-chain which contains only a non-ca type cert.

Reply:
I decoded and inspected cert-chain.pem and I believe it only contains a leaf certificate.

Reply:
hmm...  are you inspecting the one from `[samples/certs/cert-chain.pem](https://raw.githubusercontent.com/istio/istio/master/samples/certs/cert-chain.pem)`  ?   tried `openssl x509 -in samples/certs/cert-chain.pem -noout -text` and it gives 

'X509v3 Basic Constraints:
    CA:TRUE'

Reply:
Ah, you're right, I was looking at the wrong `cert-chain.pem` file. The above test case https://github.com/istio/istio/pull/42980/files#diff-9eb462ac51166aef4fdb4ca1198d23865c737c5a1a2bd8a7b4f8da9f6e1bab27R79-R81 uses the `cert-chain.pem` as an empty cert chain or leaf certificate, but it is a CA. Is that correct?